File: src/python/src/game.py
import random

from colorama import Fore, init


init(autoreset=True)


class Game:
    def __init__(self) -> None:
        self.secret_number: int
        self.game_over = False
        self.attempts = 0
        self.max_attempts = 10

    def start(self):
        self.secret_number = random.randint(1, 100)
        print(f'{Fore.GREEN}\n\nДобро пожаловать в игру "Угадай число"')
        print(f"{Fore.BLUE}Игра загадала число от 1 до 100. Попробуйте угадать ;)")
        print(f"{Fore.BLUE}У вас есть {self.max_attempts} попыток\n")

    def guess(self, number: int):
        if number == self.secret_number:
            print(
                f"{Fore.GREEN}Поздравляем! Вы угадали число {self.secret_number} с {self.attempts} попытки"
            )
            self.game_over = True
        elif number < self.secret_number:
            print(f"{Fore.RED}Не угадали. Загаданное число больше")
        else:
            print(f"{Fore.RED}Не угадали. Загаданное число меньше")

    def run(self):
        self.start()

        while not self.game_over:
            try:
                num = int(
                    input(
                        f"[{self.attempts}/{self.max_attempts}] Введите ваше предположение: "
                    )
                )
            except ValueError:
                print(f"{Fore.RED}Введите число")
                continue

            if self.attempts < self.max_attempts:
                self.guess(num)
                self.attempts += 1
            else:
                print(f"{Fore.RED}\nУпс. Попытки закончились :(")
                self.game_over = True

        print(f"{Fore.MAGENTA}\n\nРазработчик: https://www.github.com/0xM4LL0C")
        print(
            f"{Fore.MAGENTA}Репозиторий: https://www.github.com/0xM4LL0C/guess-the-number"
        )

File: src/cpp/src/game.cpp
#include <iostream>
#include "game.h"
#include "termcolor/termcolor.hpp"
#include "GRand/grand.h"


void Game::start() {
    GRand rand;
    secretNumber = rand.i(100);
    std::cout << termcolor::green << "\n\nДобро пожаловать в игру \"Угадай число\"" << std::endl;
    std::cout << termcolor::blue << "Игра загадала число от 1 до 100. Попробуйте угадать ;)\n\n" << std::endl;
    std::cout << "У вас есть " << maxAttempts << " попыток" << termcolor::reset << std::endl;
}

void Game::guess(int number) {
    if (number == secretNumber) {
        std::cout << termcolor::green << "Поздравляем! Вы угадали число " << secretNumber << " с " << attempts << " попытки";
        gameOver = true;
    } else if (number < secretNumber) {
        std::cout << termcolor::red << "Не угадали. Загаданное число больше";
    } else {
        std::cout << termcolor::red << "Не угадали. Загаданное число меньше";
    }

    std::cout << termcolor::reset << std::endl;
}

int Game::run() {
    start();
    while (!gameOver) {
        int guessNumber;
        std::cout << "[" << attempts << "/" << maxAttempts << "] " << "Введите ваше предположение: ";
        std::cin >> guessNumber;
        if (attempts < maxAttempts) {
            guess(guessNumber);
            attempts++;
        } else {
            std::cout << termcolor::red << "Упс. Попытки закончились :(" << std::endl;
            return 1;
        }
    }

    std::cout << std::endl;
    std::cout << termcolor::magenta << "Разработчик: https://www.github.com/0xM4LL0C" << std::endl;
    std::cout << "Репозиторий: https://www.github.com/0xM4LL0C/guess-the-number" << termcolor::reset << std::endl;
    return 0;
}
